code/game/objects/items/devices/magnetic_lock.dm

```diff
@@ -402,8 +429,12 @@
 					return
 				if (1 to 4)
 					add_overlay("overlay_deconstruct_[constructionstate]")
+			if(istype(src, /obj/item/device/magnetic_lock/keypad))
```

Comment:
`istype(src, ...)` is an anti-pattern. Do `/obj/item/device/magnetic_lock/keypad/update_overlays()` instead.

Reply:
wait you already do. why is this here then?